### PR TITLE
Make the fetcher tests more resilient.

### PR DESCRIPTION
**Pull request by ligthyear**
We replace most URLs (as they break too often) with a more resilient URL-Object that checks only the URL can be parsed and specific elements are exactly as expected. This way our tests don't break all the time when they just replace a picture URL here or there...

@ezmiller please review!

### CODE REVIEW

File: beavy_modules/url_extractor/lib/fetching_test.py [hunk 1]
```diff
@@ -40,31 +41,132 @@ def __eq__(self, other):
...
+
+    def password_check(self, inp):
+        self._password_check = inp
+        return self
+
+    def params_check(self, inp):
+        self._params_check = inp
+        return self
+
+    def __eq__(self, other):
```
Comment (ezmiller) [inline beavy_modules/url_extractor/lib/fetching_test.py]: How very Judith Butler of you, Ben!

Comment (ligthyear) [inline beavy_modules/url_extractor/lib/fetching_test.py]: What? You mean the `__eq__`? That's the python [rich comparision function name](https://docs.python.org/3.4/reference/datamodel.html#object.__eq__)  to check whether one instance equals another. We used it already before in that file in `Contains` and such.

**Comment by ezmiller** [comment]
@ligthyear this looks good to me, however, there are a few things about the code that elude me. how is it that you can put the URL() class constructor directly into the JSON in the test like that? Where is that URL object instantiated and its attributes set? 

I'm going to merge this as I think these are just things I don't fully understand in python or something. 

@homu r+

File: beavy_modules/url_extractor/lib/fetching_test.py (hunk 1 above)
Comment (ezmiller) [inline beavy_modules/url_extractor/lib/fetching_test.py]: Just a joke... I didn't realize the use of self/other is key, but it makes sense. I like it!

**Comment by ligthyear** [comment]
@ezmiller 

If you don't define a `__init__` method, the default from `object` is inherited, which doesn't do anything. We have that `URL` class (or an instance of) primarily for the `__eq__` method, which allows us to inspect the other item and return whether we accept that or not (by returning true or false) – this isn't its intended use but fine for our case.

Now in the "JSON" you are refering to, that is just a Python dict and we can run any code in there inline. In that way `URL` isn't any different than the previous `Contains` methods.

But if you only did that, the `URL` would in the __eq__ call only check whether it can parse the other through `parseurl`. We want to look and ensure them a little more and that is what all the helper functions are for. When they are called, they set an instance attribute to the given value, like `_scheme`. Looking closely you'll find that our `__eq__` method has support to check those if set against the others value (and the `_check` functions). Thus we can easily extend (or configure as you said) our URL check by calling `scheme("http")` on it. Now these function are _chainable_, meaning they returned the changed self. So that we can add multiple of them in one line nicely `.scheme("http").hostname("example.com")`

That is really all. Looks more fancy than it really is ;)

**Pull request merged**